pkg/devfile/parser/data/common/component_helper.go

```diff
@@ -23,3 +24,26 @@ func IsVolume(component v1.Component) bool {
 	}
 	return false
 }
+
+// GetComponentType returns the component type of a given component
+func GetComponentType(component v1.Component) (v1.ComponentType, error) {
+	if component.Container != nil {
+		return v1.ContainerComponentType, nil
+	}
+	if component.Volume != nil {
+		return v1.VolumeComponentType, nil
+	}
+	if component.Plugin != nil {
+		return v1.PluginComponentType, nil
+	}
+	if component.Kubernetes != nil {
+		return v1.KubernetesComponentType, nil
+	}
+	if component.Openshift != nil {
+		return v1.OpenshiftComponentType, nil
+	}
+	if component.Custom != nil {
+		return v1.CustomComponentType, nil
+	}
+	return "", fmt.Errorf("Unknown component type")
```

Comment:
err should not be capital. I think we can also opt for a switch case here instead of if blocks